app/models/profile.rb

```diff
@@ -117,6 +126,10 @@ def fix_http str
 
   delegate :provider_id_for_audits, :to => :account, :allow_nil => true
 
+  def will_save_change_to_logo?
+    logo.dirty?
```

Comment:
I'm not entirely sure about the logic behind this `dirty?` method but, by checking the usage in this class, couldn't we use `ActiveRecord::AttributeMethods::Dirty`'s `has_changes_to_save?` or even `will_save_change_to_attribute?(:logo)`?

Reply:
`logo` is not a normal attribute. We set some DB fields and paperclip uses them to process `logo`.

https://stackoverflow.com/questions/9029002/how-to-detect-if-a-paperclip-attachment-was-changed-in-after-save-callback

Reply:
It is not:
```
[1] pry(main)> Profile.take.logo.has_changes_to_save?
NoMethodError: undefined method `has_changes_to_save?' for #<Paperclip::Attachment:0x0000000007c523b0>
```